tools/compileProtos.ts

```diff
@@ -104,24 +105,45 @@ async function buildListOfProtos(protoJsonFiles: string[]): Promise<string[]> {
  * @param {string[]} protos List of proto files to compile.
  */
 async function compileProtos(protos: string[]): Promise<void> {
-  const output = path.join('protos', 'protos.json');
+  // generate protos.json file from proto list
+  const jsonOutput = path.join('protos', 'protos.json');
   if (protos.length === 0) {
     // no input file, just emit an empty object
-    await writeFile(output, '{}');
+    await writeFile(jsonOutput, '{}');
     return;
   }
-  const pbjsArgs = [
+  const pbjsArgs4JSON = [
     '--target',
     'json',
     '-p',
     path.join(__dirname, '..', '..', 'protos'),
     '-p',
     'protos',
     '-o',
-    output,
+    jsonOutput,
   ];
-  pbjsArgs.push(...protos);
-  await pbjsMain(pbjsArgs);
+  pbjsArgs4JSON.push(...protos);
+  await pbjsMain(pbjsArgs4JSON);
+
+  // generate protos/protos.js from protos.json
+  const jsOutput = path.join('protos', 'protos.js');
+  const pbjsArgs4js = [
+    '--target',
+    'static-module',
+    '-p',
+    path.join(__dirname, '..', '..', 'protos'),
+    '-p',
+    'protos',
+    '-o',
+    jsOutput,
+  ];
+  pbjsArgs4js.push(...protos);
+  await pbjsMain(pbjsArgs4js);
+
+  // generate protos/protos.d.ts
+  const tsOuput = path.join('protos', 'protos.d.ts');
```

Comment:
typo, should be `tsOutput` (`t` is missing)

Reply:
you are right, thanks!